lib/gcloud/vision/image.rb

```diff
@@ -359,8 +356,9 @@ def inspect
       ##
       # @private The Google API Client object for the Image.
       def to_gapi
-        if content?
-          { content: content }
+        if io?
+          @io.rewind
+          { content: Base64.strict_encode64(@io.read) }
```

Comment:
Not really. Calculating cryptographic hashes are different than encoding the IO data.

The Vision image object needs to read the entire string in order to encode it, and then the encoded string is given to the API call. Not every image object's IO object will be persisted on the filesystem, some could be a StringIO object which are already in memory.

But once the image object reads from the IO and encrypts to Base64, the image object no longer holds on to a reference to the contents, and so the contents can be garbage collected. Previously the image object put the contents into an ivar and so the contents stayed around as long as the vision image object did.